.github/workflows/ci-build-staging.yml

```diff
@@ -0,0 +1,52 @@
+name: Build-Staging
+on:
+  push:
+    branches:
+      - staging
+  pull_request:
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version-file: .node-version
+      - name: Use Yarn Cache
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/yarn
+          key: yarn-${{ hashFiles(format('{0}{1}', github.workspace, '/yarn.lock')) }}
+          restore-keys: yarn-
+      - name: Install
+        run: yarn
+      - name: Lint
+        run: yarn lint
+      - name: Type Check
+        run: yarn type-check
+      - name: Format
+        run: yarn format:check
+      - name: Test
+        env:
+          FIREBASE_PRIVATE_KEY: ${{ secrets.FIREBASE_PRIVATE_KEY }}
+          FIREBASE_PRIVATE_KEY_ID: ${{ secrets.FIREBASE_PRIVATE_KEY_ID }}
+          USE_PROD_DB: false
+          NEXT_PUBLIC_IS_STAGING: true
```

Comment:
Not sure if it's possible but oculd base the values off of `github.base_ref` 
https://docs.github.com/en/actions/learn-github-actions/contexts.

And then don't need two separate files.

Reply:
I used [this hack](https://github.com/orgs/community/discussions/25725) to try to set the env vars with ternary operators, but we'll see if it works

Reply:
Currently trying [this one](https://dev.to/ku6ryo/how-to-set-env-parameters-conditionally-in-github-actions-1g6a)